Expanded coefficients and Added broadcasting for * and /

Vizia128 opened the pull request:
Added broadcasting to allow for inertial maps in PGA physics.
```julia
Base.broadcasted(::typeof(*), a::MultiVector, b::MultiVector)::MultiVector
Base.broadcasted(::typeof(/), a::MultiVector, b::MultiVector)::MultiVector
```
Added the ability to grab a tuple or vector of coefficients from a MultiVector. This allows for grabbing the coefficients of a sub algebra.
```julia
coefficients(mv::MultiVector, idxs::Union{NTuple, AbstractVector})
```

Fixed a bug where isapprox(0*alg.e1, 0*alg.e1) returns false.

Vizia128 committed:
Expanded coefficients and Added broadcasting for * and /

ATell-SoundTheory commented:
Thank you very much for your contribution. I've done a quick code review and I have a few minor issues:

- Documentation string says the function name is `coefficient`, but the code says `coefficients`.
- You do not specify the value type for the vector or tuple handed into 'coefficients', but you're then assuming it's an index type and not a symbol. Since the `coefficient` function you're generalizing handles both, you should, too. You are also relying on more internals than you have to. You should probably just forward the array/tuple to the already defined coefficient functions with a map.
- I like the way you've constructed the submitted unit tests. However, the broadcast test is rather hard to maintain and quite elaborate. Since you want to replicate component-wise operations, why do you not construct the expected outcome from known component-wise operations in Julia instead of spelling them out with numeric literals?

If you address these issues there shouldn't be anything in the way of merging your changes.

Vizia128 committed:
Added symbol input for coefficients, improved tests, assert BIa in BIb

Vizia128 commented:
Thank you for the feedback. I think I have addressed the issues.
- Fixed docstring of `coefficients` to be plural
- I added support for inputting vectors or ntuples of symbols into `coefficients`
- Simplified the unit tests and separated the .* and ./ tests
- Added an assertion for mv1 ./ mv2 to ensure that mv1's base indices are a subset of mv2's
